### PR TITLE
172723 dev ops(cd) activities for iat

### PR DESCRIPTION
**Pull request by nerellavinod**
Created CD pipeline for IAT environment.
Azure resources created ( Webapp, Appinsight, KeyValut & EventHub ) using Terraform scripts and deployed on IAT environment.

### CODE REVIEW

File: azure-pipelines.yml
Comment (amogh14893) [inline azure-pipelines.yml:1]: Could you please check if variable **ResourceGroup** is being used anywhere else please have it removed from all deployment stages.

**Review by amogh14893 — CHANGES_REQUESTED** [review]
one minor review comment . Please validate

File: azure-pipelines.yml
Comment (nerellavinod) [reply azure-pipelines.yml:1]: resolved comment

**Review by amogh14893 — APPROVED** [review]
LGTM . Approved

File: azure-pipelines.yml [hunk 1]
```diff
@@ -451,8 +449,55 @@ stages:
...
+            deploy:
+              steps:
+                - template: Deployment/templates/continuous-deployment-app.yml
+                  parameters:
+                    AzureSubscription: "S-100 Permit Generator vNext-E2E - A.011.15.5"
+  
+  - stage: IATDeploy
+    dependsOn:  Devdeploy
+    displayName: "IATdeploy (inc terraform, webapp deploy)"
+    condition: and(succeeded(), or(eq(variables['Build.SourceBranch'], 'refs/heads/main'),startsWith(variables['Build.SourceBranch'], 'refs/heads/release/'), eq(variables['Build.SourceBranch'], 'refs/heads/develop')))
```
Comment (jrippington) [inline azure-pipelines.yml:467]: Don't think this is correct - you must not be able to run any branches based on 'develop' into IAT (or PreProd). The rules should be:

dev - all branches
iat - only branches based on main, not develop

vnext iat - only branches based on develop, not main
vmext e2e - only branches based on develop, not main

Could you please review this (and other stages) to ensure we're correct as above?

**Review by jrippington — CHANGES_REQUESTED** [review]
One point to check the stage gate rule logic please.

File: azure-pipelines.yml (hunk 1 above)
Comment (nerellavinod) [reply azure-pipelines.yml]: resolved comment.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -372,7 +370,7 @@ stages:
   - stage: vNextIATDeploy
     dependsOn:  Devdeploy
     displayName: "vNext IATdeploy (inc terraform, webapp deploy)"
-    condition: and(succeeded(), or(eq(variables['Build.SourceBranch'], 'refs/heads/main'),startsWith(variables['Build.SourceBranch'], 'refs/heads/release/'), eq(variables['Build.SourceBranch'], 'refs/heads/develop')))    
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/develop'))
```
Comment (jrippington) [inline azure-pipelines.yml:373]: Should this not be 'startswith'?

**Review by jrippington — CHANGES_REQUESTED** [review]
Quick query!

File: azure-pipelines.yml (hunk 2 above)
Comment (nerellavinod) [reply azure-pipelines.yml:373]: As discussed , current changes suffice the above requirement.

**Pull request merged**